File: android-sdk-ui/src/main/java/com/braze/ui/inappmessage/factories/DefaultInAppMessageSlideupViewFactory.java
package com.braze.ui.inappmessage.factories;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;

import com.appboy.ui.R;
import com.braze.Braze;
import com.braze.enums.BrazeViewBounds;
import com.braze.images.IBrazeImageLoader;
import com.braze.models.inappmessage.IInAppMessage;
import com.braze.models.inappmessage.InAppMessageSlideup;
import com.braze.support.BrazeLogger;
import com.braze.support.StringUtils;
import com.braze.ui.inappmessage.IInAppMessageViewFactory;
import com.braze.ui.inappmessage.views.InAppMessageBaseView;
import com.braze.ui.inappmessage.views.InAppMessageSlideupView;
import com.braze.ui.support.ViewUtils;

public class DefaultInAppMessageSlideupViewFactory implements IInAppMessageViewFactory {
  private static final String TAG = BrazeLogger.getBrazeLogTag(DefaultInAppMessageSlideupViewFactory.class);

  @Override
  public InAppMessageSlideupView createInAppMessageView(@NonNull Activity activity, @NonNull IInAppMessage inAppMessage) {
    InAppMessageSlideupView view = (InAppMessageSlideupView) activity.getLayoutInflater().inflate(R.layout.com_braze_inappmessage_slideup, null);
    if (ViewUtils.isDeviceNotInTouchMode(view)) {
      BrazeLogger.w(TAG, "The device is not currently in touch mode. This message requires user touch interaction to display properly.");
      return null;
    }

    InAppMessageSlideup inAppMessageSlideup = (InAppMessageSlideup) inAppMessage;
    Context applicationContext = activity.getApplicationContext();
    view.applyInAppMessageParameters(inAppMessage);

    String imageUrl = InAppMessageBaseView.getAppropriateImageUrl(inAppMessageSlideup);
    if (!StringUtils.isNullOrEmpty(imageUrl)) {
      IBrazeImageLoader appboyImageLoader = Braze.getInstance(applicationContext).getImageLoader();
      appboyImageLoader.renderUrlIntoInAppMessageView(applicationContext, inAppMessage, imageUrl, view.getMessageImageView(), BrazeViewBounds.IN_APP_MESSAGE_SLIDEUP);
    }

    view.setMessageBackgroundColor(inAppMessageSlideup.getBackgroundColor());
    view.setMessage(inAppMessageSlideup.getMessage());
    view.setMessageTextColor(inAppMessageSlideup.getMessageTextColor());
    view.setMessageTextAlign(inAppMessageSlideup.getMessageTextAlign());
    view.setMessageIcon(inAppMessageSlideup.getIcon(), inAppMessageSlideup.getIconColor(), inAppMessageSlideup.getIconBackgroundColor());
    view.setMessageChevron(inAppMessageSlideup.getChevronColor(), inAppMessageSlideup.getClickAction());
    view.resetMessageMargins(inAppMessageSlideup.getImageDownloadSuccessful());
    return view;
  }

}

File: android-sdk-ui/src/main/java/com/braze/ui/inappmessage/factories/DefaultInAppMessageSlideupViewFactory.kt
package com.braze.ui.inappmessage.factories

import android.app.Activity
import com.braze.ui.R
import com.braze.Braze
import com.braze.enums.BrazeViewBounds
import com.braze.models.inappmessage.IInAppMessage
import com.braze.models.inappmessage.InAppMessageSlideup
import com.braze.support.BrazeLogger.Priority.W
import com.braze.support.BrazeLogger.brazelog
import com.braze.ui.inappmessage.IInAppMessageViewFactory
import com.braze.ui.inappmessage.views.InAppMessageBaseView
import com.braze.ui.inappmessage.views.InAppMessageSlideupView
import com.braze.ui.support.isDeviceNotInTouchMode

open class DefaultInAppMessageSlideupViewFactory : IInAppMessageViewFactory {
    override fun createInAppMessageView(
        activity: Activity,
        inAppMessage: IInAppMessage
    ): InAppMessageSlideupView? {
        val view = activity.layoutInflater.inflate(
            R.layout.com_braze_inappmessage_slideup,
            null
        ) as InAppMessageSlideupView
        if (isDeviceNotInTouchMode(view)) {
            brazelog(W) { "The device is not currently in touch mode. This message requires user touch interaction to display properly." }
            return null
        }
        val inAppMessageSlideup = inAppMessage as InAppMessageSlideup
        val applicationContext = activity.applicationContext
        view.applyInAppMessageParameters(inAppMessage)
        val imageUrl = InAppMessageBaseView.getAppropriateImageUrl(inAppMessageSlideup)
        if (!imageUrl.isNullOrEmpty()) {
            val brazeImageLoader = Braze.getInstance(applicationContext).imageLoader
            view.messageImageView?.let {
                brazeImageLoader.renderUrlIntoInAppMessageView(
                    applicationContext,
                    inAppMessage,
                    imageUrl,
                    it,
                    BrazeViewBounds.IN_APP_MESSAGE_SLIDEUP
                )
            }
        }
        view.setMessageBackgroundColor(inAppMessageSlideup.backgroundColor)
        inAppMessageSlideup.message?.let { view.setMessage(it) }
        view.setMessageTextColor(inAppMessageSlideup.messageTextColor)
        view.setMessageTextAlign(inAppMessageSlideup.messageTextAlign)
        inAppMessageSlideup.icon?.let {
            view.setMessageIcon(
                it,
                inAppMessageSlideup.iconColor,
                inAppMessageSlideup.iconBackgroundColor
            )
        }
        view.setMessageChevron(inAppMessageSlideup.chevronColor, inAppMessageSlideup.clickAction)
        view.resetMessageMargins(inAppMessageSlideup.imageDownloadSuccessful)
        return view
    }
}
